app/src/main/java/com/sun/music_66/view/widget/CollapsibleToolbar.kt

```diff
@@ -0,0 +1,20 @@
+package com.sun.music_66.view.widget
+
+import android.content.Context
+import android.util.AttributeSet
+import androidx.constraintlayout.motion.widget.MotionLayout
+import com.google.android.material.appbar.AppBarLayout
+
+class CollapsibleToolbar @JvmOverloads constructor(
+    context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) :
+    MotionLayout(context, attrs, defStyleAttr), AppBarLayout.OnOffsetChangedListener {
+
+    override fun onOffsetChanged(appBarLayout: AppBarLayout?, verticalOffset: Int) {
+        progress = -verticalOffset / appBarLayout?.totalScrollRange?.toFloat()!!
```

Comment:
theo em, anh viết như này thì đoạn ?. kia sẽ không có ý nghĩa gì. Nó null vẫn bị dính NullPointerException
```suggestion
        appBarLayout?.totalScrollRange?.let { total ->
            progress = -verticalOffset / total.toFloat()
        }
```
hoặc 
```suggestion
        progress = appBarLayout?.totalScrollRange?.let { total ->
            -verticalOffset / total.toFloat()
        } ?: DEFAULT_VALUE
```